modules/tensor_mechanics/src/userobjects/CrystalPlasticityStateVarRateComponentVoce.C

```diff
@@ -0,0 +1,271 @@
+                                     "12-23 = 2.0 and 24-48 = 3.0 "
+                                     " usually these are all 1.");
+  params.addParam<std::vector<Real>>("coplanarHardening_group_values",
+                                     "The coplanar laten hardening coefficient q_ab"
+                                     "correspoinding to each group"
+                                     " i.e. '1.0 2.0 3.0' means 0-11 = 1.0, "
+                                     "12-23 = 2.0 and 24-48 = 3.0 ");
+  params.addParam<std::vector<Real>>("GroupGroup_Hardening_group_values",
+                                     "The group-to-group laten hardening coefficient q_ab"
+                                     "This is a NxN vector"
+                                     " i.e. '1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0' "
+                                     "means non-coplanar slip systems in gr_11,22,33= "
+                                     "1.0, 5.0 and 9.0 respectively."
+                                     "latent ahrdening between for gr_12,13 = 2.0 3.0"
+                                     " respectively");
+  params.addClassDescription(
+      "Phenomenological Voce constitutive model state"
+      " variable evolution rate "
+      "component base class.  Override this virtual functions in your class");
+  return params;
+}
+
+CrystalPlasticityStateVarRateComponentVoce::CrystalPlasticityStateVarRateComponentVoce(
+    const InputParameters & parameters)
+  : CrystalPlasticityStateVarRateComponent(parameters),
+    _mat_prop_slip_rate(
+        getMaterialProperty<std::vector<Real>>(parameters.get<std::string>("uo_slip_rate_name"))),
+    _mat_prop_state_var(
+        getMaterialProperty<std::vector<Real>>(parameters.get<std::string>("uo_state_var_name"))),
+    _crystal_lattice_type(getParam<MooseEnum>("crystal_lattice_type")),
+    _groups(getParam<std::vector<unsigned int>>("groups")),
+    _h0_group_values(getParam<std::vector<Real>>("h0_group_values")),
+    _tau0_group_values(getParam<std::vector<Real>>("tau0_group_values")),
+    _tauSat_group_values(getParam<std::vector<Real>>("tauSat_group_values")),
+    _hardeningExponent_group_values(getParam<std::vector<Real>>("hardeningExponent_group_values")),
+    _selfHardening_group_values(getParam<std::vector<Real>>("selfHardening_group_values")),
+    _coplanarHardening_group_values(getParam<std::vector<Real>>("coplanarHardening_group_values")),
+    _GroupGroup_Hardening_group_values(
+        getParam<std::vector<Real>>("GroupGroup_Hardening_group_values"))
+{
+  // perform input checks and initialize usefull variables
+  _n_groups = _groups.size() - 1;
+  checkHardeningParametersSize();
+  initSlipSystem_PlaneID(_slipSystem_PlaneID);
+  initSlipSystem_GroupID(_slipSystem_GroupID);
+}
+
+void
+CrystalPlasticityStateVarRateComponentVoce::checkHardeningParametersSize() const
+{
+  // check that at least one group exists
+  if (_n_groups <= 0)
```

Comment:
`_n_groups` can never be < 0. It being 0 means only one group was specified. Is that invalid?

Reply:
Daniel, thanks a lot for spotting this mistake. I copy and pasted this from CrystalPlasticityStateVariable.C without double checking the code. Therefore this file also need to be checked. 

Anyway, Following the previous comment , this check is now
if (_n_groups < 2), because for one group to be defined we need at least two values.